lib/streamlit/server/Server.py

```diff
@@ -86,6 +89,44 @@ class State(Enum):
     STOPPED = "STOPPED"
 
 
+class ExceededRetries(Exception):
+    pass
+
+
+def server_port_is_manually_set():
+    return config.is_manually_set("server.port")
+
+
+def start_listening(app, call_count=0):
+    """makes the server start listening at the configured port.
```

Comment:
Pydoc styling:
- Capitalize the first line
- Add a blank line between the first line and the rest.
- Put the closing `"""` on its own line.